paddle/fluid/framework/dlpack_tensor.cc

```diff
@@ -120,5 +121,39 @@ DLPackTensor::DLPackTensor(const Tensor &tensor, LaneType lanes) {
   t_.byte_offset = 0;
 }
 
+::DLManagedTensor *DLPackTensor::toDLManagedTensor() {
+  // init shape, tensor dims
+  // for DLManagedTensor shape need to be compatible with ndim
+  // refer to cupy and cudf, we new int64[ndim]
+  auto shape = new int64_t[t_.ndim];
+  using DimType = decltype(t_.ndim);  // int
+  for (DimType i = 0; i < t_.ndim; ++i) {
+    shape[i] = t_.shape[i];
+  }
+  t_.shape = shape;
+
+  // init strides, nullptr means the tensor is compact
+  // refer to cupy and cudf, the compact tensor first dim's strides need to be 1
+  // and second dim's strides need to be length of rows of cudf
+  // cudf now only support dim=2
+  if (t_.ndim > 1)
```

Comment:
Seems that cudf but not `DLManagedTensor` only supports dim=2. Therefore, should we add an `PADDLE_ENFORCE_LT(t_.ndim, 2)` here or change the function name from `toDLManagedTensor` to something like `toCudfCompatibleDLManagedTensor`?

Reply:
Done.